File: src/main/java/com/travel/domain/travel/domain/plan/TravelPlanRepository.java
package com.travel.domain.travel.domain.plan;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;

import java.util.List;

public interface TravelPlanRepository  extends JpaRepository<TravelPlan, Long> {
    TravelPlan findTopByTravelIdAndTravelDateOrderByOrderNoDesc(Long travelId, String TravelDate);
    List<TravelPlan> findByTravelId(Long travelId);
    List<TravelPlan> findByTravelIdAndTravelDate(Long travelId, String travelDate);
    @Modifying(clearAutomatically = true, flushAutomatically = true)
    void deleteAllByTravelIdAndTravelDate(Long travelId, String TravelDate);

    Long countByTravelIdAndTravelDate(Long travelId, String travelDate);

}

File: src/main/kotlin/com/travel/domain/travel/domain/plan/TravelPlanRepository.kt
package com.travel.domain.travel.domain.plan

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying

interface TravelPlanRepository: JpaRepository<TravelPlan, Long> {
    fun findTopByTravelIdAndTravelDateOrderByOrderNoDesc(travelId: Long, travelDate: String): TravelPlan
    fun findByTravelId(travelId: Long): List<TravelPlan>
//    fun findByTravelIdAndTravelDate(travelId: Long, travelDate: String): List<TravelPlan>

    @Modifying(clearAutomatically = true, flushAutomatically = true)
    fun deleteAllByTravelIdAndTravelDate(travelId: Long, travelDate: String)

    fun countByTravelIdAndTravelDate(travelId: Long, travelDate: String): Long
}